### DISCUSSION TITLE
Plugins: use a default plugin to match all applications (such as git-crypt)

### DISCUSSION THREAD

**Discussion by ebuildy** [argoproj/argo-cd#12604] [General]
With argocd 2.6, what is the best practice to use tools like git-crypt or sops ?

(we have tried both sidecar or CM plugins, without success, ending up with replacing git binary by a script shell to do the decrypt operation after a git fetch).

**Comment by madhatter215** [comment]
Here's an example sops plugin for CMP -> https://gist.github.com/madhatter215/f2f3a7a43e6f28103948f20c8918a472
Note: must `cat` the rest of the files that are not encrypted, nowhere in the documentation mentions it.
